src/attributes/codegen.md

```diff
@@ -140,6 +140,25 @@ feature detection on the x86 platforms.
 > may be enabled or disabled for an entire crate with the
 > [`-C target-feature`] flag.
 
+## The `track_caller` attribute
+
+The `track_caller` attribute applies to function declarations and indicates to the compiler that the
+attributed function's generated code should have an implicit `&'static core::panic::Location`
+argument which is populated by its caller. If the caller of the attributed function is also
+annotated with `#[track_caller]`, then the attributed function should receive the location
+its caller received, and so on.
+
+An implementation of `core::intrinsics::caller_location` returns this implicit argument if called
+from a function with the attribute. Otherwise it returns the location at which the intrinsic itself
+was called.
+
+Creating a function pointer to a function with `#[track_caller]` creates a shim which passes the
+location at which the attributed function is defined. The pointer then refers to this shim,
+preventing unsoundness or the implicit ABI from leaking into the function pointer's type.
+
+In a const context, the attribute does not affect function ABI and instead informs the const
+evaluator to continue walking up the stack when probing for the topmost caller location.
```

Comment:
This doesn't seem right... I did some investigation and filed https://github.com/rust-lang/rust/issues/68789.

Reply:
Thanks for the followup, I think there's some confusion (aside from the legit bugs) here caused by me incorrectly including ABI in the definition of the attribute. From miri's perspective there isn't an observable ABI change, I'll clarify the details in a rustc-specific note as you suggested above.

Reply:
Looks like I was using an outdated nightly instead of master, so there's no bug. I've filed https://github.com/rust-lang/rust/pull/68798 to ensure the correct behavior remains.